### PR TITLE
MSC2457: Invalidating devices during password modification

### PR DESCRIPTION
**Pull request by clokep**
[Rendered](https://github.com/matrix-org/matrix-doc/blob/clokep/modify-password-invalid-sessions/proposals/2457-password-modification-invalidating-devices.md)

### CODE REVIEW

File: proposals/2457-password-modification-invalidating-devices.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+An optional field is added to the JSON body body of the [password reset endpoint](https://matrix.org/docs/spec/client_server/r0.6.0#post-matrix-client-r0-account-password)
+called `logout_devices`. This is a boolean flag (defaulting to `true`) that
+signals to whether other devices and sessions should be invalidated after
+modifying the password.
+
+## Potential issues
+
+While Synapse defaults to the behavior of invalidating other devices and
+sessions this may not be true of other implementations. Thus, a default of
```
Comment (uhoreg) [inline proposals/2457-password-modification-invalidating-devices.md]: The spec currently says:

> The homeserver SHOULD NOT revoke the access token provided in the request, **however all other access tokens for the user should be revoked if the request succeeds**.

(emphasis added)  So any compliant implementations should currently invalidate other devices, so defaulting to `true` should be safe.

Comment (clokep) [reply proposals/2457-password-modification-invalidating-devices.md]: Oops, you're correct! Not sure how I missed that. I believe that removes any of the potential issues then! 👍

Comment (clokep) [reply proposals/2457-password-modification-invalidating-devices.md]: Thanks! I rewrote this section in light of this new information.

File: proposals/2457-password-modification-invalidating-devices.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+1. Modify a password and log all other devices out (for use when an account has
+   been compromised).
+2. Modify a password and do not touch any session data (for use in a
+   non-malicious situations).
+
+Alternately a client may default to whichever workflow is best for their users.
+
+## Proposal
+
+An optional field is added to the JSON body body of the [password reset endpoint](https://matrix.org/docs/spec/client_server/r0.6.0#post-matrix-client-r0-account-password)
```
Comment (anoadragon453) [inline proposals/2457-password-modification-invalidating-devices.md]:
```suggestion
An optional field is added to the JSON body of the [password reset endpoint](https://matrix.org/docs/spec/client_server/r0.6.0#post-matrix-client-r0-account-password)
```

**Comment by richvdh** [comment]
@mscbot fcp merge

**Comment by richvdh** [comment]
Team member @richvdh has proposed to merge this. The next step is review by the rest of the tagged people:

* [ ] @KitsuneRal
* [ ] @anoadragon453
* [ ] @ara4n
* [ ] @dbkr
* [ ] @erikjohnston
* [x] @richvdh
* [ ] @turt2live
* [ ] @uhoreg

Concerns:

None listed.

**Comment by richvdh** [comment]
(this is implemented in https://github.com/matrix-org/synapse/issues/6976)

File: proposals/2457-password-modification-invalidating-devices.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+* ... probably tons of others ...
+
+These can be summarized into two groups:
+
+1. "My account has been compromised and I need to re-secure it."
+2. "I just want to change my password."
+
+The [current Matrix specification](https://matrix.org/docs/spec/client_server/r0.6.0#post-matrix-client-r0-account-password)
+does not provide a way to differentiate between these use cases. It gives no
+guidance into what should happen to other sessions / devices when a password is
```
Comment (turt2live) [inline proposals/2457-password-modification-invalidating-devices.md]: It does give guidance, but also leaves it up to the implementation:

> The homeserver SHOULD NOT revoke the access token provided in the request, however all other access tokens for the user should be revoked if the request succeeds.

Comment (clokep) [reply proposals/2457-password-modification-invalidating-devices.md]: Ugh, this was noted above in https://github.com/matrix-org/matrix-doc/pull/2457#discussion_r391903274. I didn't realize there was two places this said that.

**Comment by turt2live** [comment]
Looks like @mscbot is stuck, so manually starting the process here.

Per the [checkboxes above](https://github.com/matrix-org/matrix-doc/pull/2457#issuecomment-601246845) this MSC has entered final comment period.

**Comment by turt2live** [comment]
Per the [review above](https://github.com/matrix-org/matrix-doc/pull/2457#issuecomment-601246845) this has finished final comment period!

**Comment by turt2live** [comment]
Merged 🎉 

Spec PR was https://github.com/matrix-org/matrix-doc/pull/2523

**Pull request merged**